cybox/test/__init__.py

```diff
@@ -14,6 +17,89 @@
 logger = logging.getLogger(__name__)
 
 
+@contextlib.contextmanager
+def isolation():
+    """Context manager that attempts to reset the sys.modules and globals()
+    to its initial, pre-`import cybox` state.
+
+    This helps eliminate issues where tests pass because of residuals caused
+    by previous imports.
+
+    Any code which leverages this context manager will need to re-import all
+    dependencies within.
+    """
+    def del_sys_modules(to_del):
```

Comment:
Definitely! I probably use nested functions more than I should :-\ Aside from the closure/decorator stuff, I tend to use them when a large function benefits from being broken up into blocks of repeated code, but the resulting sub-functions don't make much sense or have value outside of the parent context (nothing else calls them). I definitely struggle with the decision between nesting and breaking out a function in those cases!